packages/compass-export-to-language/src/components/modal.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import jsBeautify from 'js-beautify';
+import {
+  InfoModal,
+  css,
+  Label,
+  Code,
+  Checkbox,
+  Banner,
+  spacing,
+  FormFieldContainer,
+} from '@mongodb-js/compass-components';
+import type { Language } from '@mongodb-js/compass-components';
+import { modalOpenChanged } from '../modules/modal-open';
+import {
+  outputLanguageToCodeLanguage,
+  codeLanguageToOutputLanguage,
+} from '../modules/languages';
+import type { OutputLanguage } from '../modules/languages';
+import { getInputExpressionMode, runTranspiler } from '../modules/transpiler';
+import type { InputExpression } from '../modules/transpiler';
+
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { countAggregationStagesInString } from '../modules/count-aggregation-stages-in-string';
+const { track } = createLoggerAndTelemetry('COMPASS-EXPORT-TO-LANGUAGE-UI');
+
+type LanguageOption = {
+  displayName: string;
+  language: Language;
+};
+
+const languageOptions: LanguageOption[] = [
+  { displayName: 'Java', language: 'java' },
+  { displayName: 'Node', language: 'javascript' },
+  { displayName: 'C#', language: 'cs' },
+  { displayName: 'Python', language: 'python' },
+  { displayName: 'Ruby', language: 'ruby' },
+  { displayName: 'Go', language: 'go' },
+  { displayName: 'Rust', language: 'rust' },
+  { displayName: 'PHP', language: 'php' },
+];
+
+const shellLanguageOptions: LanguageOption[] = [
+  { displayName: 'Shell', language: 'javascript' },
+];
+
+type ExportToLanguageState = {
+  modalOpen: boolean;
+  inputExpression: InputExpression;
+  uri: string;
+  namespace: string;
+};
+
+const bannerStyles = css({
+  marginBottom: spacing[3],
+});
+
+const formFieldContainerStyles = css({
+  display: 'flex',
+  gap: spacing[4],
+});
+
+const fieldStyles = css({
+  flex: '1 1 0px',
+});
+
+const editorHeadingStyles = css({
+  marginBottom: spacing[2],
+  display: 'block',
+});
+
+const codeStyles = css({
+  alignItems: 'start',
+  height: `${spacing[6] * 4 - spacing[5]}px`,
+});
+
+const checkboxStyles = css({
+  marginTop: spacing[2],
+  // prevent an extra couple of pixels that always causes the modal to scroll unnecessarily
+  overflow: 'hidden',
```

Comment:
```suggestion
  width: 'auto',
```
Looks like this unintentionally cuts off the focus state view of the checkbox:
<img width="235" alt="Screen Shot 2022-11-07 at 12 53 03 PM" src="https://user-images.githubusercontent.com/1791149/200380788-f611f330-2d03-4b62-8331-1bc175f0f994.png">
Using `width: auto` overrides LeafyGreen's width while keeping things in view:
<img width="235" alt="Screen Shot 2022-11-07 at 12 53 56 PM" src="https://user-images.githubusercontent.com/1791149/200380963-611f2013-5681-47e8-99d5-164b68bee928.png">

Reply:
Thanks!